### ISSUE TITLE
plpython support

### ISSUE THREAD

**Issue by PhilipTrauner** [darold/pgFormatter#56]
```sql
CREATE OR REPLACE FUNCTION plpython_demo()
RETURNS void
AS $$
	from this import s, d

	for char in s:
		print(d.get(char, char), end="")

	print()
$$
LANGUAGE 'plpython3u';
```
↓
```sql
CREATE OR REPLACE FUNCTION plpython_demo ()
    RETURNS void
AS $$
FROM
    this import s,
    d FOR char IN s: print (d.get (char,
            char),
END = "") print ()
$$
LANGUAGE 'plpython3u';
```
This issue looks pretty much just like #54. It would be great if the contents of `plpython3u` and `plpythonu` language blocks could be ignored by pgFormatter.

With kind regards,
Philip Trauner

**Comment by darold** [comment]
pgFormatter is a SQL and PLPSQL code formatter, it can not format python or other external language. To solves your problem you must instruct pgFormatter that the lines between your code body separator must not be parsed. In the example above it can be done using command line:
```
pg_format -S '$$' filetoparse.sql
```
the resulting output will be:
```
CREATE OR REPLACE FUNCTION plpython_demo ()
    RETURNS void
AS $$
	from this import s, d

	for char in s:
		print(d.get(char, char), end="")

	print()
$$
    LANGUAGE 'plpython3u';
```
Code between both `$$` is kept untouched. I will see if I can remove the extra indentation before LANGUAGE. 

Actually this feature is only available at command line, not on the online website. I will add the feature to the CGI.

**Comment by PhilipTrauner** [comment]
Thanks for the quick reply!

```sql
CREATE OR REPLACE FUNCTION foo ()
    RETURNS TRIGGER
AS $$
BEGIN
    IF NEW.role NOT IN ( SELECT rolname FROM pg_authid) THEN
        RAISE EXCEPTION 'role % does not exist.', NEW.role;
    END IF;
END;
$$
LANGUAGE 'plpgsql';

CREATE OR REPLACE FUNCTION plpython_demo ()
    RETURNS void
AS $$
	from this import s, d

	for char in s:
		print(d.get(char, char), end="")

	print()
$$
LANGUAGE 'plpython3u';
```
↓
```sql
CREATE OR REPLACE FUNCTION foo ()
    RETURNS TRIGGER
AS $$
BEGIN
    IF NEW.role NOT IN ( SELECT rolname FROM pg_authid) THEN
        RAISE EXCEPTION 'role % does not exist.', NEW.role;
    END IF;
END;
$$
LANGUAGE 'plpgsql';

CREATE OR REPLACE FUNCTION plpython_demo ()
    RETURNS void
AS $$
	from this import s, d

	for char in s:
		print(d.get(char, char), end="")

	print()
$$
LANGUAGE 'plpython3u';
```

The solution you suggested sadly doesn't work in `.sql` files with multiple function languages. I'm not asking for the Python code to be formatted correctly, as that would definitely be out of scope for a Postgres formatter, instead I'm suggesting to leave all unsupported language blocks as is.

With kind regards,
Philip Trauner

**Comment by darold** [comment]
I know, the problem is that pgFormatter parse the file token by token. It would be easy if we can detect that the function have LANGUAGE set to plpython or more generally other language than SQL or PLPGSQL.

This require a preprocessing to autodetect the function language before the tokenizer. I think it is possible and I will work on it when I found some more time.

Best regards,

**Comment by darold** [comment]
Fixed in commit f0edc10.

**Comment by PhilipTrauner** [comment]
Awesome, thank you very much! Do you accept donations?  😉 

With kind regards,
Philip Trauner
